File: rewrite.py
import re

with open('lib/core/l10n/app_strings.dart', 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
for line in lines:
    if 'static const' in line and '//' in line:
        match = re.search(r"static const\s+(\w+)\s*=\s*'([^']+)';\s*//\s*(.*)", line)
        if match:
            name = match.group(1)
            amh = match.group(2)
            eng = match.group(3).strip()
            # Handle cases where the english translation is just descriptive and not an exact translation,
            # or where we need to escape quotes.
            eng = eng.replace("'", "\\'")
            new_lines.append(f"  static String get {name} => isAmharic ? '{amh}' : '{eng}';\n")
            continue
    new_lines.append(line)

new_lines.insert(6, "  static bool isAmharic = true;\n\n")

with open('lib/core/l10n/app_strings.dart', 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

File: rewrite.js
const fs = require('fs');

const content = fs.readFileSync('lib/core/l10n/app_strings.dart', 'utf-8');
const lines = content.split('\n');

const newLines = [];
let inserted = false;

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    
    if (!inserted && line.includes('abstract class S {')) {
        newLines.push(line);
        newLines.push('  static bool isAmharic = true;');
        newLines.push('');
        inserted = true;
        continue;
    }

    if (line.includes('static const') && line.includes('//')) {
        const match = line.match(/static const\s+(\w+)\s*=\s*'([^']+)';\s*\/\/\s*(.*)/);
        if (match) {
            const name = match[1];
            const amh = match[2];
            let eng = match[3].trim();
            eng = eng.replace(/'/g, "\\'");
            newLines.push(`  static String get ${name} => isAmharic ? '${amh}' : '${eng}';`);
            continue;
        }
    }
    newLines.push(line);
}

fs.writeFileSync('lib/core/l10n/app_strings.dart', newLines.join('\n'), 'utf-8');
